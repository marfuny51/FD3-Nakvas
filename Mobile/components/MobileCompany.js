import React from 'react';
import PropTypes from 'prop-types';

import MobileClient from './MobileClient';

import './MobileCompany.css';

class MobileCompany extends React.PureComponent {

  static propTypes = {
    name: PropTypes.string.isRequired,
    clients:PropTypes.array.isRequired,
  };

  state = {
    name: this.props.name,
    clients: this.props.clients,
  };

  setName1 = () => {
    this.setState({name:'МТС'});
  };

  setName2 = () => {
    this.setState({name:'Velcom'});
  };
  
  render() {

    console.log("MobileCompany render");

    var clientsCode=this.state.clients.map( client =>
      <MobileClient key={client.id} clients={client}  />
    );

    return (
      <div className='MobileCompany'>
        <input type="button" value="Velcom" onClick={this.setName1} />
        <input type="button" value="MTS" onClick={this.setName2} />
        <div className='MobileCompanyName'>Company:{this.state.name}</div><br/>
        <input type="button" value="All" />
        <input type="button" value="Active" />
        <input type="button" value="Blocked" /><br/><br/>
        <table className='AllTable'>
          <thead className='Title'>
              <tr>
                  <th>Surname</th>
                  <th>Name</th>
                  <th>Middle name</th>
                  <th>Balance</th>
                  <th>Status</th>
                  <th>Edit</th>
                  <th>Delete</th>
              </tr>
          </thead>
          <tbody className='Table'>
          {clientsCode}
          </tbody>
      </table><br/>
        <input type="button" value="Add new client" />
      </div>
    )
    ;

  }

}

export default MobileCompany;
