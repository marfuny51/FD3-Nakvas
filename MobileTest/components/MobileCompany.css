th, td{
    text-align: center; 
    padding: 5px; 
    border: 1px solid black; 
    border-collapse: collapse;
}

table {
    border-collapse: collapse;  
}