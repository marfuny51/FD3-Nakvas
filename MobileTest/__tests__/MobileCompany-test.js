"use strict";

import React from 'react';
import MobileCompany from '../components/MobileCompany';
import renderer from 'react-test-renderer';

test('работа MobileCompany', () => {

  let clientsArr=[ 
    {id:101, surname:'Ivanov', name: 'Ivan', otch: 'Ivanovich', balance:200}, 
    {id:102, surname:'Sidorov', name: 'Sidor', otch: 'Sidorovich', balance:250}, 
    {id:103, surname:'Petrov', name: 'Petr', otch: 'Petrovich', balance:180},
    {id:104, surname:'Grigoriev', name: 'Grigoriy', otch: 'Grigorievich', balance:-220},
  ];
  // создаём тестовую версию компонента
  const component = renderer.create(
    <MobileCompany 
    clients={clientsArr}
  />
  );

  // получаем снэпшот (HTML-снимок) компонента для сверки, что вёрстка не испортилась
  let componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();

  // найдём в вёрстке компонента кнопку фильтрации 'All'
  const buttonFiltrAll = component.root.find( el => el.props.value == 'All'); 
  // и "нажмём" на неё
  buttonFiltrAll.props.onClick();

  // найдём в вёрстке компонента кнопку фильтрации 'Active'
  const buttonFiltrActive = component.root.find( el => el.props.value=='Active'); 
  // и "нажмём" на неё
  buttonFiltrActive.props.onClick();

  // найдём в вёрстке компонента кнопку фильтрации 'Blocked'
  const buttonFiltrBlocked = component.root.find( el => el.props.value=='Blocked'); 
  // и "нажмём" на неё
  buttonFiltrBlocked.props.onClick();

  // найдём в вёрстке компонента кнопку фильтрации 'Add new client'
  const buttonAdd = component.root.find( el => el.props.value=='Add new client'); 
  // и "нажмём" на неё
  buttonAdd.props.onClick();

  // найдём в вёрстке компонента кнопку фильтрации 'Edit'
  //const buttonEdit = component.root.find( el => el.props.value=='Edit'); 
  // и "нажмём" на неё
  //buttonEdit.props.onClick();

  // найдём в вёрстке компонента кнопку фильтрации 'Delete'
  //const buttonDelete = component.root.find( el => el.props.value=='Delete'); 
  // и "нажмём" на неё
  //buttonDelete.props.onClick();

  // получаем уже изменённый снэпшот
  componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();

  // "нажмём" кнопки ещё раз
  buttonFiltrAll.props.onClick();
  buttonFiltrActive.props.onClick();
  buttonFiltrBlocked.props.onClick();
  buttonAdd.props.onClick();
  //buttonEdit.props.onClick();
  //buttonDelete.props.onClick();
  
  // и получаем окончательный снэпшот
  componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();
  
  /*
  // можно эмулировать события, передавая в качестве объекта события то что нам нужно:
  wrapper.find('select').simulate('change', {
    target: { value: "hello" },
  });
  */

});
