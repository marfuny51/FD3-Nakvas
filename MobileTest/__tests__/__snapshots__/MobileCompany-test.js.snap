// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`работа MobileCompany 1`] = `
<div
  className="MobileCompany"
>
  <input
    onClick={[Function]}
    type="button"
    value="All"
  />
  <input
    onClick={[Function]}
    type="button"
    value="Active"
  />
  <input
    onClick={[Function]}
    type="button"
    value="Blocked"
  />
  <br />
  <br />
  <table
    className="AllTable"
  >
    <thead
      className="Title"
    >
      <tr>
        <th>
          Surname
        </th>
        <th>
          Name
        </th>
        <th>
          Middle name
        </th>
        <th>
          Balance
        </th>
        <th>
          Status
        </th>
        <th>
          Edit
        </th>
        <th>
          Delete
        </th>
      </tr>
    </thead>
    <tbody
      className="Table"
    >
      <tr>
        <td>
          Ivanov
        </td>
        <td>
          Ivan
        </td>
        <td>
          Ivanovich
        </td>
        <td>
          200
        </td>
        <td
          className="ActiveClient"
        >
          active
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Edit"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Delete"
          />
        </td>
      </tr>
      <tr>
        <td>
          Sidorov
        </td>
        <td>
          Sidor
        </td>
        <td>
          Sidorovich
        </td>
        <td>
          250
        </td>
        <td
          className="ActiveClient"
        >
          active
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Edit"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Delete"
          />
        </td>
      </tr>
      <tr>
        <td>
          Petrov
        </td>
        <td>
          Petr
        </td>
        <td>
          Petrovich
        </td>
        <td>
          180
        </td>
        <td
          className="ActiveClient"
        >
          active
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Edit"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Delete"
          />
        </td>
      </tr>
      <tr>
        <td>
          Grigoriev
        </td>
        <td>
          Grigoriy
        </td>
        <td>
          Grigorievich
        </td>
        <td>
          -220
        </td>
        <td
          className="BlockedClient"
        >
          blocked
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Edit"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Delete"
          />
        </td>
      </tr>
    </tbody>
  </table>
  <br />
  <input
    onClick={[Function]}
    type="button"
    value="Add new client"
  />
</div>
`;

exports[`работа MobileCompany 2`] = `
<div
  className="MobileCompany"
>
  <input
    onClick={[Function]}
    type="button"
    value="All"
  />
  <input
    onClick={[Function]}
    type="button"
    value="Active"
  />
  <input
    onClick={[Function]}
    type="button"
    value="Blocked"
  />
  <br />
  <br />
  <table
    className="AllTable"
  >
    <thead
      className="Title"
    >
      <tr>
        <th>
          Surname
        </th>
        <th>
          Name
        </th>
        <th>
          Middle name
        </th>
        <th>
          Balance
        </th>
        <th>
          Status
        </th>
        <th>
          Edit
        </th>
        <th>
          Delete
        </th>
      </tr>
    </thead>
    <tbody
      className="Table"
    >
      <tr>
        <td>
          Grigoriev
        </td>
        <td>
          Grigoriy
        </td>
        <td>
          Grigorievich
        </td>
        <td>
          -220
        </td>
        <td
          className="BlockedClient"
        >
          blocked
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Edit"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Delete"
          />
        </td>
      </tr>
    </tbody>
  </table>
  <br />
  <input
    onClick={[Function]}
    type="button"
    value="Add new client"
  />
  <div
    className="EditAdd"
  >
    <span>
      ID: 
    </span>
    <span>
      6
    </span>
    <br />
    <span>
      Surname: 
    </span>
    <input
      defaultValue=""
      type="text"
    />
    <br />
    <span>
      Name: 
    </span>
    <input
      defaultValue=""
      type="text"
    />
    <br />
    <span>
      Middle name: 
    </span>
    <input
      defaultValue=""
      type="text"
    />
    <br />
    <span>
      Balance: 
    </span>
    <input
      defaultValue=""
      type="text"
    />
    <br />
    <input
      onClick={[Function]}
      type="button"
      value="Save"
    />
    <input
      onClick={[Function]}
      type="button"
      value="Cancel"
    />
  </div>
</div>
`;

exports[`работа MobileCompany 3`] = `
<div
  className="MobileCompany"
>
  <input
    onClick={[Function]}
    type="button"
    value="All"
  />
  <input
    onClick={[Function]}
    type="button"
    value="Active"
  />
  <input
    onClick={[Function]}
    type="button"
    value="Blocked"
  />
  <br />
  <br />
  <table
    className="AllTable"
  >
    <thead
      className="Title"
    >
      <tr>
        <th>
          Surname
        </th>
        <th>
          Name
        </th>
        <th>
          Middle name
        </th>
        <th>
          Balance
        </th>
        <th>
          Status
        </th>
        <th>
          Edit
        </th>
        <th>
          Delete
        </th>
      </tr>
    </thead>
    <tbody
      className="Table"
    >
      <tr>
        <td>
          Grigoriev
        </td>
        <td>
          Grigoriy
        </td>
        <td>
          Grigorievich
        </td>
        <td>
          -220
        </td>
        <td
          className="BlockedClient"
        >
          blocked
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Edit"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Delete"
          />
        </td>
      </tr>
    </tbody>
  </table>
  <br />
  <input
    onClick={[Function]}
    type="button"
    value="Add new client"
  />
  <div
    className="EditAdd"
  >
    <span>
      ID: 
    </span>
    <span>
      6
    </span>
    <br />
    <span>
      Surname: 
    </span>
    <input
      defaultValue=""
      type="text"
    />
    <br />
    <span>
      Name: 
    </span>
    <input
      defaultValue=""
      type="text"
    />
    <br />
    <span>
      Middle name: 
    </span>
    <input
      defaultValue=""
      type="text"
    />
    <br />
    <span>
      Balance: 
    </span>
    <input
      defaultValue=""
      type="text"
    />
    <br />
    <input
      onClick={[Function]}
      type="button"
      value="Save"
    />
    <input
      onClick={[Function]}
      type="button"
      value="Cancel"
    />
  </div>
</div>
`;
